best-practices/native/react-native/common-components/buttons/button.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { StyleSheet, TouchableOpacity, Text, Image } from "react-native";
+import PropTypes from "prop-types";
+
+interface ButtonProps {
+  onPress: Function;
+  text: string;
+  icon: any;
```

Comment:
How do you feel about getting specific on the `icon` field? 

Something like: https://react-native-elements.github.io/react-native-elements/docs/icon.html

Reply:
I think Johnny left this ambiguous so that any icon from the various icon sets available could be used, but I wouldn't mind being more specific.

Reply:
I should probably ask, is there a standard Icon object within react native (like the above link) or is it just an arbitrary object?

Reply:
Nope, there is no standard. We _could_ make an Interface or Type with the common properties between them all.

Reply:
I think ATM this is fine. Making an interface would mean less leeway later on. And I think we should support different kinds of Icons. 

If we find this to be an issue in the future or a sign on confusion later with others that use the components, we can always document and change.